cvFit/src/app/macroCalculator/calculator.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component } from "@angular/core";
+interface activityLevel{
+    value: number;
+    viewValue: string;
+}
+interface weightGoal{
+    value: number;
+    viewValue: string;
+}
+@Component({
+    selector: 'calculator-component',
+    templateUrl: './calculator.component.html',
+    styleUrls: ['./calculator.component.css']
+})
+export class CalculatorComponent{
+    //variables for calculations
+    age: number | undefined;
+    gender: string | undefined; 
+    genders: string[] = ["Male", "Female"];
+    heightF: number | undefined;
+    heightI: number | undefined;
+    weight: number | undefined;
+    activity: number = 1; //part of dropdown
+    activityL: activityLevel[] = [{value: 1, viewValue: "No exercise"}, {value: 2, viewValue: "Exercise 1-3 times a week"}, {value: 3, viewValue: "Exercise 4-5 times a week"}, {value: 4, viewValue: "Exercise 6-7 times a week"}]
+    goal: number = 1; //part of dropdown
+    goalW: weightGoal[] = [{value: 1, viewValue: "Weight loss"}, {value: 2, viewValue: "Maintain weight"}, {value: 3, viewValue: "Gain weight"}]
+
+    //variables for results 
+    protein: number =0;
+    carbs: number =0;
+    fat: number =0;
+    dailyCals: number | undefined;
+    
+    showResult: boolean = false;
+    bmr: number | undefined;
+    
+    submit(formValues: { age: number; gender: string; heightF: number; heightI: number; weight: number; activity: number; goal: number; }){
+        
+        this.showResult = true;
+        this.age = formValues.age;
+        this.gender = formValues.gender;
+        this.heightF = formValues.heightF;
+        this.heightI = formValues.heightI;
+        this.weight = formValues.weight;
+        
+        if (this.gender == 'Male'){
```

Comment:
All of these calculations are great for unit testing. You should look into unit testing soon and see if you can come back and ensure that we're doing the right calcuations on each of these values. This may also show that these calculations probably shouldn't be here. They should be in a service level so that you can easier test it and share it across the application.